### DISCUSSION TITLE
How to set Homebrew to compile for a particular architecture?

### DISCUSSION THREAD

**Discussion by randomshinichi** [Homebrew/discussions#648] [Everyday usage]
I want to build bottles for several CPU architectures in Tigerbrew. The [available ones are](https://github.com/mistydemeo/tigerbrew/blob/fd341dd94006c10c4d8717a647cb0c9f63021e8a/Library/Homebrew/os/mac/hardware.rb#L8)
```
    :g3 => "-mcpu=750",
    :g4 => "-mcpu=7400",
    :g4e => "-mcpu=7450",
    :g5 => "-mcpu=970",
```
1. How can I build a bottle without installing it?
2. Now I have a G5, but how can I tell it "okay, compile the G3 optimized version" and then the G4 one, and then save them as different bottles?
3. Where are bottles saved anyway? I got this far but it should at least mention where it's saved.
```
pmg5:Formula shinichi$ brew reinstall openssl.rb --build-bottle
==> Reinstalling openssl
==> Downloading https://www.openssl.org/source/openssl-1.1.1i.tar.gz
Already downloaded: /Users/shinichi/Library/Caches/Homebrew/openssl-1.1.1i.tar.gz
==> perl ./Configure --prefix=/usr/local/Cellar/openssl/1.1.1i --openssldir=/usr/local/etc/openssl no-ssl3 no-ssl3-method no-zlib darwin-ppc-cc
==> make depend
==> make
==> make test
==> make install MANDIR=/usr/local/Cellar/openssl/1.1.1i/share/man MANSUFFIX=ssl
==> Not running post_install as we're building a bottle
You can run it manually using `brew postinstall openssl`
==> Caveats
A CA file has been bootstrapped using certificates from the SystemRoots
keychain. To add additional certificates (e.g. the certificates added in
the System keychain), place .pem files in
  /usr/local/etc/openssl/certs

and run
  /usr/local/opt/openssl/bin/c_rehash
==> Summary
/usr/local/Cellar/openssl/1.1.1i: 1195 files, 46M, built in 11.0 minutes
```

**Comment by SMillerDev** [comment]
> How can I build a bottle without installing it?

You can't, building is always part of an install step.
>Now I have a G5, but how can I tell it "okay, compile the G3 optimized version" and then the G4 one, and then save them as different bottles?

You can't with a default homebrew install as far as I know. We always compile on the target architecture.
> Where are bottles saved anyway? I got this far but it should at least mention where it's saved.

Try running with `--debug`

**Comment by carlocab** [comment]
>How can I build a bottle without installing it?

I don't think that's possible. To build a bottle, you would do something like `brew install --build-bottle $formula` then `brew bottle $formula`.

>Now I have a G5, but how can I tell it "okay, compile the G3 optimized version" and then the G4 one, and then save them as different bottles?

I think you would need to hack `brew`'s source to do this.

>Where are bottles saved anyway? I got this far but it should at least mention where it's saved.

Try `cd $(brew --cache)`.
